langchain/evaluation/run_evaluators/agent_trajectory_run_evaluator.py

```diff
@@ -0,0 +1,294 @@
+"""Run evaluator wrapper for string evaluators."""
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, List, Optional, Tuple, TypedDict
+
+from langchainplus_sdk import EvaluationResult, RunEvaluator
+from langchainplus_sdk.schemas import Example, Run
+from pydantic import Field
+
+from langchain.agents.agent import AgentExecutor
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForChainRun,
+    CallbackManagerForChainRun,
+)
+from langchain.chains.base import Chain
+from langchain.evaluation.run_evaluators.utilities.trajectory_utils import (
+    assemble_agent_trajectory,
+)
+from langchain.evaluation.schema import AgentTrajectoryEvaluator
+from langchain.load.serializable import Serializable
+from langchain.schema import RUN_KEY
+from langchain.schema.agent import AgentAction
+
+logger = logging.getLogger(__name__)
+
+
+class AgentInput(TypedDict):
+    """Values to pass to the trajectory evaluator."""
+
+    input: str
+    prediction: str
+    agent_trajectory: List[Tuple[AgentAction, str]]
+    reference: Optional[str]
+
+
+class AgentTrajectoryRunMapper(Serializable):
+    """Extract Agent Trajectory (and inputs and prediction) to be evaluated."""
+
+    input_key: str = "input"
+    """The key from the model Run's inputs to use as the eval input."""
+    prediction_key: str = "output"
+    """The key from the model Run's outputs to use as the eval prediction."""
+
+    @property
+    def output_keys(self) -> List[str]:
+        """The keys to extract from the run."""
+        return ["prediction", "input", "agent_trajectory"]
+
+    @classmethod
+    def from_chain(
+        cls,
+        model: Chain,
+        input_key: Optional[str] = None,
+        prediction_key: Optional[str] = None,
+    ) -> AgentTrajectoryRunMapper:
+        """Create a RunMapper from a chain."""
+        error_messages = []
```

Comment:
I don't love how much duplicate code there is here. Could extract some functions since I'm not convinced inheritance is really the right thing here